cmds/core/dd/dd.go

```diff
@@ -63,21 +62,24 @@ var (
 	bytesWritten int64 // access atomically, must be global for correct alignedness
 )
 
-const (
-	convNoTrunc = 1 << iota
-	oFlagSync
-	oFlagDSync
-)
+type setclear struct {
+	set   int
+	clear int
+}
 
-var convMap = map[string]int{
-	"notrunc": convNoTrunc,
+// N.B. The flags in os derive from syscall.
+// They are, hence, mutually exclusive, on target
+// kernels.
+var convMap = map[string]setclear{
+	"notrunc": {clear: os.O_TRUNC},
 }
 
-var flagMap = map[string]int{
-	"sync":  oFlagSync,
-	"dsync": oFlagDSync,
+var flagMap = map[string]setclear{
+	"sync": {set: os.O_SYNC},
 }
 
+var permflags = os.O_TRUNC | os.O_SYNC
```

Comment:
permflags sounds like permanentFlags. allowedFlags? permittedFlags if you spell it out?

Reply:
allowedFlags